aten/src/ATen/native/TensorIterator.h

```diff
@@ -189,6 +193,17 @@ struct CAFFE2_API TensorIterator {
     return operands_[arg].tensor;
   }
 
+  void cast_outputs() {
+    if (compute_common_dtype_) {
+      for(int i=0; i < noutputs(); i++) {
+        if (operands_[i].original_buffer.defined() && dtype(i) != operands_[i].original_buffer.scalar_type()) {
+          operands_[i].original_buffer.copy_(operands_[i].tensor);
+          operands_[i].tensor = operands_[i].original_buffer;
```

Comment:
why do you do this step?

Reply:
`operands_[i].tensor` is the one that will actually be returned by a call to `output()`